backends/advanced/src/advanced_omi_backend/controllers/system_controller.py

```
@@ -1059,3 +966,187 @@
+        from advanced_omi_backend.memory_config_loader import MemoryConfigLoader
+        
+        # Parse YAML
+        try:
+            parsed_config = yaml.safe_load(config_yaml)
+            if not parsed_config:
+                return JSONResponse(
+                    status_code=400, content={"error": "Configuration file is empty"}
+                )
+        except yaml.YAMLError as e:
+            return JSONResponse(
+                status_code=400, content={"error": f"Invalid YAML syntax: {str(e)}"}
```

Ruff (0.12.2)

Use explicit conversion flag

Replace with conversion flag

(RUF010)